lib/rspec/core/formatters.rb

```diff
@@ -170,6 +170,11 @@ def add(formatter_to_use, *paths)
       end
     end
 
+    # @private
+    def change_output_stream(output_stream)
+      @formatters.each { |formatter| formatter.change_output(output_stream) }
```

Comment:
Given that this only works for our formatters and ones that inherit ours, it makes me wonder if a different way of doing this would be better.  One idea is to not pass the formatters a raw `output` object but instead a wrapper object that holds a reference to a real `output` internally.  If the `formatter_loader` also has a reference to the same `output_wrapper` object that each formatter is given, the `formatter_loader` can tell the `output_wrapper` to change where it outputs to, and all the formatters should "just work" with no changes needed.  Thoughts?